### PR TITLE
use dialog element instead of window.confirm

### PR DESCRIPTION
**Pull request by bafolts**
This should work to replace `window.confirm` with a dialog element.

image

Moved more logic into `methods` from the `mounted` function as well. Eventually we could probably work this all onto the board component.

### CODE REVIEW

File: src/components/SelectSpace.ts [hunk 1]
```diff
@@ -21,111 +22,114 @@ export const SelectSpace = Vue.component('select-space', {
...
+        return board.getElementsByClassName('board-space-selectable');
+      }
+      throw new Error('main board not found!');
+    },
+    onTileSelected: function(tile: HTMLElement) {
+      this.selectedTile = tile;
+      this.disableAvailableSpaceAnimation();
+      this.animateSpace(tile, true);
+      tile.classList.remove('board-space--available');
       const hideTileConfirmation = PreferencesManager.loadValue('hide_tile_confirmation') === '1';
```
Comment (kberg) [inline src/components/SelectSpace.ts:86]: How much extra work would it take to include the "hide" checkbox I prpopsed in the bug? That's the real goal.

Comment (bafolts) [reply src/components/SelectSpace.ts:86]: For you? No extra work at all! I will add it. It isn't work when you enjoy it so technically this is no work at all.

Comment (bafolts) [reply src/components/SelectSpace.ts:91]: @kberg I created a `ConfirmDialog` component for re-use. Attached new screenshot. The "hide" checkbox is now there.

File: src/components/SelectSpace.ts [hunk 2]
```diff
@@ -21,111 +21,111 @@ export const SelectSpace = Vue.component('select-space', {
...
       if (hideTileConfirmation) {
-        return true;
+        this.confirmPlacement();
+      } else {
+        (this.$refs['confirmation'] as any).show();
```
Comment (bafolts) [inline src/components/SelectSpace.ts:95]: I want to remove this `any`. Meant to do this before I pushed.

Comment (kberg) [reply src/components/SelectSpace.ts:95]: Well I sure won't stop you.

**Review by kberg — APPROVED** [review]
Looks great upon visual inspection. I might play with it beforehand, but it hits the right notes.

**Pull request merged**